Bring up iOS 17, tvOS 17, watchOS 10, and Sonoma builders and testers

robert-jenner opened the pull request:
#### 3fa725d246c1b46d37c38777ac2fb47106d8369a
<pre>
Bring up iOS 17, tvOS 17, watchOS 10, and Sonoma builders and testers
rdar://116012814
<a href="https://bugs.webkit.org/show_bug.cgi?id=262065">https://bugs.webkit.org/show_bug.cgi?id=262065</a>

Reviewed by NOBODY (OOPS!).

Configuration change for bring up of Sonoma and friends.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOS16.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOS16Simulator.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOS17Simulator.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOSDevice.png:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOSSimulator.png:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/Sonoma.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS16.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS16Simulator.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS17.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS17Simulator.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/Ventura.png:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOS10.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOS9.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOSSimulator10.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOSSimulator9.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/iOS17.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Dashboard.js:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/WebKitBuildbot.js:
(WebKitBuildbot):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Styles/Main.css:
(table.queue-grid tr.platform.ios-simulator-17 img.logo):
(table.queue-grid tr.platform.ios-17 img.logo):
(table.queue-grid tr.platform.tvos-simulator-17 img.logo):
(table.queue-grid tr.platform.tvos-17 img.logo):
(table.queue-grid tr.platform.watchos-simulator-10 img.logo):
(table.queue-grid tr.platform.watchos-10 img.logo):
(table.queue-grid tr.platform.macos-sonoma img.logo):
(table.queue-grid tr.platform.ios-simulator-16 img.logo): Deleted.
(table.queue-grid tr.platform.ios-16 img.logo): Deleted.
(table.queue-grid tr.platform.tvos-simulator-16 img.logo): Deleted.
(table.queue-grid tr.platform.tvos-16 img.logo): Deleted.
(table.queue-grid tr.platform.watchos-simulator-9 img.logo): Deleted.
(table.queue-grid tr.platform.watchos-9 img.logo): Deleted.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3fa725d246c1b46d37c38777ac2fb47106d8369a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/19863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/20286 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20905 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/21754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/18551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/20098 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/23543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/20432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/21754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/20083 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/23543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/17271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/22608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/23543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/18066 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/22608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/17267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/23543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/18241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/22608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/19229 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/18834 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/20432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/23263 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/19687 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/18006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/17902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/23263 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/22354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 jsc-mips~~](https://ews-build.webkit.org/#/builders/24/builds/24517 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/18659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/24/builds/24517 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

robert-jenner committed:
Bring up iOS 17, tvOS 17, watchOS 10, and Sonoma builders and testers
rdar://116012814
https://bugs.webkit.org/show_bug.cgi?id=262065

Reviewed by NOBODY (OOPS!).

Configuration change for bring up of Sonoma and friends.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOS16.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOS16Simulator.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOS17Simulator.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOSDevice.png:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/IOSSimulator.png:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/Sonoma.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS16.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS16Simulator.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS17.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/TvOS17Simulator.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/Ventura.png:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOS10.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOS9.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOSSimulator10.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/WatchOSSimulator9.png: Removed.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Images/iOS17.png: Added.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Dashboard.js:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/WebKitBuildbot.js:
(WebKitBuildbot):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Styles/Main.css:
(table.queue-grid tr.platform.ios-simulator-17 img.logo):
(table.queue-grid tr.platform.ios-17 img.logo):
(table.queue-grid tr.platform.tvos-simulator-17 img.logo):
(table.queue-grid tr.platform.tvos-17 img.logo):
(table.queue-grid tr.platform.watchos-simulator-10 img.logo):
(table.queue-grid tr.platform.watchos-10 img.logo):
(table.queue-grid tr.platform.macos-sonoma img.logo):
(table.queue-grid tr.platform.ios-simulator-16 img.logo): Deleted.
(table.queue-grid tr.platform.ios-16 img.logo): Deleted.
(table.queue-grid tr.platform.tvos-simulator-16 img.logo): Deleted.
(table.queue-grid tr.platform.tvos-16 img.logo): Deleted.
(table.queue-grid tr.platform.watchos-simulator-9 img.logo): Deleted.
(table.queue-grid tr.platform.watchos-9 img.logo): Deleted.

aprotyas committed:
PointerEvent.button should be -1 when mouse buttons are in released state
https://bugs.webkit.org/show_bug.cgi?id=262089
rdar://116027480

Reviewed by Wenson Hsieh.

This commit fixes part of the failure in the
pointerevent_attributes_hoverable_pointers.html?mouse WPT, where the
failure mode suggests we're returning 0 for the button attribute of a
pointer event instead of -1.

This failure reflects an inconformity with the Pointer Events W3C spec
in our implementation of the button proprerty. As specified, an event's
button property should be -1 when "Neither buttons nor touch/pen contact
changed since last event", but it does _not_ suggest limiting this
behavior to pointermove events only, which is what we were doing.

To be better aligned with other browser vendors (and the spec), we make
our button property heuristic less restrictive and applicable for all
pointer event types (barring pointerup/pointerdown, since they represent
button change). In doing so, we also take the liberty to rename
MouseEvent::PointerMove to MouseEvent::PointerHasNotChanged, since the
latter case better reflects the meaning of the `-1` button value.

* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_hoverable_pointers_mouse-expected.txt:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::button const):

* Source/WebCore/dom/PointerEvent.cpp:
(WebCore::PointerEvent::PointerEvent):
Do not unconditionally assign MouseButton::Left (0) to the button
property.

* Source/WebCore/dom/PointerEvent.h:
(WebCore::PointerEvent::typeIsUpOrDown):

Make `buttonForType` platform agnostic (and public), making it a more
useful utility and reducing code duplication.

* Source/WebCore/dom/ios/PointerEventIOS.cpp:
(WebCore::buttonForType): Deleted.
* Source/WebCore/page/PointerCaptureController.cpp:
(WebCore::pointerEventType):
(WebCore::PointerCaptureController::pointerEventForMouseEvent):
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/platform/PlatformMouseEvent.h:

Rename MouseButton::PointerMove to MouseButton::PointerHasNotChanged.

Canonical link: https://commits.webkit.org/268459@main

rniwa committed:
REGRESSION(266828@main): Nested setTimeout not being throttled in the background
https://bugs.webkit.org/show_bug.cgi?id=262086
<rdar://115902836>

Reviewed by Chris Dumez.

The bug was caused by TimerAlignment and hasReachedMaxNestingLevel state getting set after
the timer had already been scheduled. We need to update these states before we schedule it.

To do this, new variants of scheduleTask and scheduleRepeatingTask which takes TimerAlignment
and hasReachedMaxNestingLevel boolean as arguments are added and deployed them in DOMTimer.

* LayoutTests/fast/dom/timer-throttling-hidden-page-2-expected.txt: Added.
* LayoutTests/fast/dom/timer-throttling-hidden-page-2.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/dom/EventLoop.cpp:
(WebCore::EventLoop::scheduleTask):
(WebCore::EventLoop::scheduleRepeatingTask):
(WebCore::EventLoopTaskGroup::scheduleTask):
(WebCore::EventLoopTaskGroup::scheduleRepeatingTask):
(WebCore::EventLoopTaskGroup::setTimerAlignment): Deleted.
* Source/WebCore/dom/EventLoop.h:
* Source/WebCore/page/DOMTimer.cpp:
(WebCore::DOMTimer::DOMTimer):

Canonical link: https://commits.webkit.org/268460@main

ThomasDevoogdt committed:
GPUExternalTextureDescriptor: fix compilation without video support

https://bugs.webkit.org/show_bug.cgi?id=262039

Reviewed by Michael Catanzaro.

GPUVideoSource is only copmiled if ENABLE(VIDEO),
so we can't have a member variable for it.

The static mediaIdentifierForSource function calls playerIdentifier which is also not present

Signed-off-by: Thomas Devoogdt <thomas.devoogdt@barco.com>
Canonical link: https://commits.webkit.org/268461@main

ThomasDevoogdt committed:
WebProcess.h: fix PlatformDisplay is not a member of WebCore

https://bugs.webkit.org/show_bug.cgi?id=262108

Reviewed by Michael Catanzaro.

It's safe to include PlatformDisplay if no EGL, so drop this check.

Signed-off-by: Thomas Devoogdt <thomas.devoogdt@barco.com>
Canonical link: https://commits.webkit.org/268462@main

bls1999 committed:
[Gardening]: New Test (264177@main): [ macOS wk1 ] imported/w3c/web-platform-tests/html/browsers/browsing-the-web/navigating-across-documents/replace-before-load/form-submit-cross-frame.html is a flaky/consistent failure.
rdar://116066790
https://bugs.webkit.org/show_bug.cgi?id=262130

Unreviewed test gardening.

* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: https://commits.webkit.org/268463@main

achristensen07 committed:
Update SiteIsolation.ChildBeingNavigatedToMainFrameDomainByParent API test to check termination of correct process
https://bugs.webkit.org/show_bug.cgi?id=262122
rdar://116060764

Reviewed by Pascoe.

I was trying to verify the iframe process was being terminated, but I was getting the pid after the iframe had navigated
to the same domain as the main frame process, which should not be terminated during the life of the WKWebView.
This gets the correct pid, and also removes some checks that are redundant with the checkFrameTreesInProcesses call below.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/268464@main

bls1999 committed:
[Gardening]: REGRESSION(267947@main?): [ macOS wk1 release ] js/ShadowRealm-importValue.html is a flakily/consistently timing out.
rdar://116068872
https://bugs.webkit.org/show_bug.cgi?id=262133

Unreviewed test gardening.

* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: https://commits.webkit.org/268465@main

twilco committed:
AX: Don't eagerly AXIsolatedTree::updateNode if we're collecting unresolved appends
https://bugs.webkit.org/show_bug.cgi?id=262077
rdar://problem/116018776

Reviewed by Chris Fleizach.

This patch features two optimizations:

  1. Change AXIsolatedTree::updateChildrenForObjects to not resolve node
     changes between each object update.
  2. Do not eagerly AXIsolatedTree::updateNode if we're gathering
     m_unresolvedPendingAppends.

Both allow for a lot more de-duplication in m_unresolvedPendingAppends,
resulting in much fewer node changes created on some pages.

On an internal webpage, before this patch we created ~165k node
changes in response to a dynamic change. After this patch, we create
~132k node changes.

* Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp:
(WebCore::AXIsolatedTree::updateNode):
(WebCore::AXIsolatedTree::updateChildrenForObjects):

Canonical link: https://commits.webkit.org/268466@main

alanbaradlay committed:
[IFC][Ruby] Rename appendAssociatedRubyAnnotationBoxIfNeeded and annotationVerticalExtent
https://bugs.webkit.org/show_bug.cgi?id=262113

Reviewed by Antti Koivisto.

* Source/WebCore/layout/formattingContexts/inline/InlineLineBoxBuilder.cpp:
(WebCore::Layout::LineBoxBuilder::setLayoutBoundsForInlineBox const):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::InlineDisplayContentBuilder::appendInlineBoxDisplayBox):
(WebCore::Layout::InlineDisplayContentBuilder::handleInlineBoxEnd):
(WebCore::Layout::InlineDisplayContentBuilder::appendInterlinearRubyAnnotationBox):
(WebCore::Layout::InlineDisplayContentBuilder::appendAssociatedRubyAnnotationBoxIfNeeded): Deleted.
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.h:
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.cpp:
(WebCore::Layout::RubyFormattingContext::annotationContributionToLayoutBounds):
(WebCore::Layout::RubyFormattingContext::annotationVerticalExtent): Deleted.
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.h:

Canonical link: https://commits.webkit.org/268467@main

xeenon committed:
Add support for Web Extension native messaging via delegate methods.
https://webkit.org/b/262082
rdar://problem/116022790

Reviewed by Brady Eidson.

Introduces two new _WKWebExtensionControllerDelegate methods for native messaging.
The first is for single message sending via runtime.sendNativeMessage(), and the other
is for persistent messaging vis runtime.connectNative(), which also introduces the
_WKWebExtensionMessagePort class for handling the port based messaging.

The ultimate plan is to have native messaging work by default internally via NSExtension,
which is tracked by https://webkit.org/b/262081. These delegate methods will still be
useful for testing or for clients that want alternative native messaging.

* Source/WebKit/Modules/OSX_Private.modulemap:
* Source/WebKit/Modules/iOS_Private.modulemap:
* Source/WebKit/Platform/cocoa/CocoaHelpers.h:
(WebKit::isValidJSONObject):
* Source/WebKit/Platform/cocoa/CocoaHelpers.mm:
(WebKit::isValidJSONObject):
(WebKit::encodeJSONData):
* Source/WebKit/Shared/API/APIObject.h:
* Source/WebKit/Shared/Cocoa/APIObject.mm:
(API::Object::newObject):
* Source/WebKit/Shared/Extensions/WebExtensionContentWorldType.h:
(WebKit::toDebugString):
* Source/WebKit/Shared/Extensions/WebExtensionContentWorldType.serialization.in:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionControllerDelegate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMessagePort.h: Added.
(NS_ERROR_ENUM):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMessagePort.mm: Added.
(-[_WKWebExtensionMessagePort dealloc]):
(-[_WKWebExtensionMessagePort isEqual:]):
(-[_WKWebExtensionMessagePort applicationIdentifier]):
(-[_WKWebExtensionMessagePort isDisconnected]):
(-[_WKWebExtensionMessagePort sendMessage:completionHandler:]):
(-[_WKWebExtensionMessagePort disconnectWithError:]):
(-[_WKWebExtensionMessagePort _apiObject]):
(-[_WKWebExtensionMessagePort _webExtensionMessagePort]):
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMessagePortInternal.h: Added.
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionMessagePortPrivate.h: Added.
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIPermissionsCocoa.mm:
(WebKit::WebExtensionContext::permissionsRequest):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIPortCocoa.mm:
(WebKit::WebExtensionContext::portPostMessage):
(WebKit::WebExtensionContext::portDisconnect):
(WebKit::WebExtensionContext::removePort):
(WebKit::WebExtensionContext::addNativePort):
(WebKit::WebExtensionContext::removeNativePort):
(WebKit::WebExtensionContext::fireQueuedPortMessageEventsIfNeeded):
(WebKit::WebExtensionContext::sendQueuedNativePortMessagesIfNeeded):
(WebKit::WebExtensionContext::firePortDisconnectEventIfNeeded):
(WebKit::WebExtensionContext::fireQueuedPortMessageEventIfNeeded): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIRuntimeCocoa.mm:
(WebKit::WebExtensionContext::runtimeConnect):
(WebKit::WebExtensionContext::runtimeSendNativeMessage):
(WebKit::WebExtensionContext::runtimeConnectNative):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPITabsCocoa.mm:
(WebKit::WebExtensionContext::tabsConnect):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionMessagePortCocoa.mm: Added.
(WebKit::toAPI):
(WebKit::toWebExtensionMessagePortError):
(WebKit::WebExtensionMessagePort::WebExtensionMessagePort):
(WebKit::WebExtensionMessagePort::operator== const):
(WebKit::WebExtensionMessagePort::extensionContext const):
(WebKit::WebExtensionMessagePort::isDisconnected const):
(WebKit::WebExtensionMessagePort::disconnect):
(WebKit::WebExtensionMessagePort::reportDisconnection):
(WebKit::WebExtensionMessagePort::remove):
(WebKit::WebExtensionMessagePort::sendMessage):
(WebKit::WebExtensionMessagePort::receiveMessage):
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
(WebKit::WebExtensionContext::create):
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.messages.in:
* Source/WebKit/UIProcess/Extensions/WebExtensionMessagePort.h: Added.
(WebKit::WebExtensionMessagePort::create):
(WebKit::WebExtensionMessagePort::~WebExtensionMessagePort):
(WebKit::WebExtensionMessagePort::applicationIdentifier const):
(WebKit::WebExtensionMessagePort::channelIdentifier const):
(WebKit::WebExtensionMessagePort::wrapper const):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIPortCocoa.mm:
(WebKit::WebExtensionAPIPort::sender):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIRuntimeCocoa.mm:
(WebKit::WebExtensionAPIRuntime::isPropertyAllowed):
(WebKit::WebExtensionAPIRuntime::sendMessage):
(WebKit::WebExtensionAPIRuntime::sendNativeMessage):
(WebKit::WebExtensionAPIRuntime::connectNative):
(WebKit::WebExtensionContextProxy::dispatchRuntimeMessageEvent):
(WebKit::WebExtensionContextProxy::dispatchRuntimeConnectEvent):
* Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPITabsCocoa.mm:
(WebKit::WebExtensionAPITabs::sendMessage):
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIPort.h:
(WebKit::WebExtensionAPIPort::WebExtensionAPIPort):
* Source/WebKit/WebProcess/Extensions/API/WebExtensionAPIRuntime.h:
* Source/WebKit/WebProcess/Extensions/Interfaces/WebExtensionAPIRuntime.idl:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIRuntime.mm:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.h:
* Tools/TestWebKitAPI/cocoa/TestWebExtensionsDelegate.mm:
(-[TestWebExtensionsDelegate webExtensionController:sendMessage:toApplicationIdentifier:forExtensionContext:replyHandler:]):
(-[TestWebExtensionsDelegate webExtensionController:connectUsingMessagePort:forExtensionContext:completionHandler:]):

Canonical link: https://commits.webkit.org/268468@main

kmiller68 committed:
Reland Atomics in IPInt
https://bugs.webkit.org/show_bug.cgi?id=262118

Reviewed by Yusuke Suzuki.

This is a fixed patch for 268300@main and 268252@main. Before there was an issue that
we were using the `unimplementedInstruction` macro instead of the `reservedOpcode`
macro for the holes in the Atomic instruction set. The `unimplementedInstruction`
macro emits a global label, which on iOS would get stripped. I think this is happening
as the definition of the label is not in the same translation unit as the declaration,
which has the attribute, so the attribute doesn't get merged into the definition. Right
now it seems the only thing preserving the IPInt labels is the `IPInt::initialize()` function
and that function did not have any references to the Atomic intruction gaps.

* JSTests/wasm/stress/non-canonical-extended-ops.js: Added.
(instance):
* Source/JavaScriptCore/llint/InPlaceInterpreter.asm:
* Source/JavaScriptCore/llint/InPlaceInterpreter.cpp:
(JSC::IPInt::initialize):
* Source/JavaScriptCore/llint/InPlaceInterpreter.h:
* Source/JavaScriptCore/runtime/OptionsList.h:
* Source/JavaScriptCore/wasm/WasmFunctionIPIntMetadataGenerator.cpp:
(JSC::Wasm::FunctionIPIntMetadataGenerator::addLength):
* Source/JavaScriptCore/wasm/WasmFunctionIPIntMetadataGenerator.h:
* Source/JavaScriptCore/wasm/WasmFunctionParser.h:
(JSC::Wasm::FunctionParser<Context>::truncSaturated):
* Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp:
(JSC::Wasm::IPIntGenerator::addMemoryFill):
(JSC::Wasm::IPIntGenerator::addMemoryCopy):
(JSC::Wasm::IPIntGenerator::atomicLoad):
(JSC::Wasm::IPIntGenerator::atomicStore):
(JSC::Wasm::IPIntGenerator::atomicBinaryRMW):
(JSC::Wasm::IPIntGenerator::atomicCompareExchange):
(JSC::Wasm::IPIntGenerator::atomicWait):
(JSC::Wasm::IPIntGenerator::atomicNotify):
(JSC::Wasm::IPIntGenerator::atomicFence):
(JSC::Wasm::IPIntGenerator::truncSaturated):
* Source/JavaScriptCore/wasm/WasmOperationsInlines.h:
(JSC::Wasm::memoryAtomicWait32):
(JSC::Wasm::memoryAtomicWait64):
* Source/JavaScriptCore/wasm/WasmSlowPaths.cpp:
(JSC::LLInt::WASM_IPINT_EXTERN_CPP_DECL):
* Source/JavaScriptCore/wasm/WasmSlowPaths.h:

Canonical link: https://commits.webkit.org/268469@main

whsieh committed:
[iOS] [WK2] Adopt _UIAsyncDragInteraction and _UIAsyncDragInteractionDelegate
https://bugs.webkit.org/show_bug.cgi?id=262006
rdar://114331777

Reviewed by Tim Horton.

As a part of cleaning up asynchronous `UIInteraction` integration in WebKit, adopt a new
`_UIAsyncDragInteraction` subclass that encapsulates the delegate methods for both (1) starting a
drag session asynchronously, and (2) adding items to an existing drag session asynchronously.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:

Add a runtime feature flag as well, to guard adoption of these new async `UIInteraction`s in UIKit.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _dragInteractionClass]):

Add a helper method to return the drag interaction class to instantiate, based on runtime and build-
time flags.

(-[WKContentView setUpDragAndDropInteractions]):
(-[WKContentView asyncDragInteraction:prepareDragSession:completion:]):
(-[WKContentView asyncDragInteraction:itemsForAddingToSession:withTouchAtPoint:completion:]):

Implement the new `_UIAsyncDragInteractionDelegate` methods here.

Canonical link: https://commits.webkit.org/268470@main

sysrqb committed:
REGRESSION(267579@main): ASSERTION FAILED: !m_pendingNavigationID
https://bugs.webkit.org/show_bug.cgi?id=261837
rdar://115799843

Reviewed by Chris Dumez.

Before 267579@main, a same-document navigation would not have an associated
navigation ID. In that change, I removed the same-document check and now all
navigations are given a navigation ID from the start. However, that change
caused an assertion failure in WebPage because WebPage was not told when a
same-document navigation completed, and This caused a situation where WebPage
was tracking two pending navigation IDs.

This patch resolves the issue by moving the current logic from
WebLocalFrameLoaderClient::dispatchDidChangeLocationWithinPage into a new
member function WebPage::didSameDocumentNavigationForFrame and resetting
m_pendingNavigationID in that function when the WebFrame is the main frame.

I manually ran all WKBackForwardList tests in a debug build and verified that
this change resolves the assertion failure.

* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidChangeLocationWithinPage):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didSameDocumentNavigationForFrame):
* Source/WebKit/WebProcess/WebPage/WebPage.h:

Canonical link: https://commits.webkit.org/268471@main

mscottapple committed:
rdar://89218473 ([ER] WebKitTestRunner and DumpRenderTree should let the user know if loading a file or a url failed)

Reviewed by David Kilzer (ddkilzer).

This adds preflight checking for the existence of input files specified on the command line, as well as reporting of any URL loading errors.

* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(initializeGlobalsFromCommandLineOptions):
(runTest):
* Tools/DumpRenderTree/mac/ResourceLoadDelegate.h:
* Tools/DumpRenderTree/mac/ResourceLoadDelegate.mm:
(-[ResourceLoadDelegate dealloc]):
(-[ResourceLoadDelegate webView:resource:didFailLoadingWithError:fromDataSource:]):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createOtherPlatformWebView):
(WTR::TestController::createWebViewWithOptions):
(WTR::TestController::createTestURL):
(WTR::TestController::runTest):
(WTR::TestController::didFailProvisionalNavigation):
(WTR::pathSuitableForTestResult):
* Tools/WebKitTestRunner/TestController.h:

Canonical link: https://commits.webkit.org/268472@main

fujii committed:
REGRESSION(268453@main): error: incomplete type 'WebCore::Document' used in type trait expression
https://bugs.webkit.org/show_bug.cgi?id=261007

Debug builds of some ports couldn't compile. Unreviewed build fix.

> include\type_traits(1317,71): error: incomplete type 'WebCore::Document' used in type trait expression

* Source/WebKit/WebProcess/Network/WebSocketProvider.cpp:
Include <WebCore/Document.h>.

Canonical link: https://commits.webkit.org/268473@main

ThomasDevoogdt committed:
StyleAdjuster.cpp: fix has no member named isMediaElement

https://bugs.webkit.org/show_bug.cgi?id=262106

Reviewed by Michael Catanzaro.

The Element class only adds the isMediaElement function if ENABLE(VIDEO).

Signed-off-by: Thomas Devoogdt <thomas.devoogdt@barco.com>
Canonical link: https://commits.webkit.org/268474@main

ThomasDevoogdt committed:
GPUExternalTextureDescriptor: fix compilation without video support

https://bugs.webkit.org/show_bug.cgi?id=262039

Reviewed by Mike Wyrzykowski.

Commit ac630888d1faae104b5e8bed6c30078e9036abb1 accidentally readded
mediaIdentifier which was dropped in commit f9d7fea52f903441640365484bf48dc5d50d82ff.

Signed-off-by: Thomas Devoogdt <thomas@devoogdt.com>
Canonical link: https://commits.webkit.org/268475@main

smfr committed:
Introduce UnifiedPDFPlugin
https://bugs.webkit.org/show_bug.cgi?id=262095
rdar://116029997

Reviewed by Tim Horton.

Flesh out the interface of PDFPluginBase, which is a pure virtual base class for the
old PDFPlugin and the new UnifiedPDFPlugin. These may get renamed later.

Move code which we are sure to share to PDFPluginBase, and add stubs in UnifiedPDFPlugin.
This is all mechanical, other than some minor cleanup the the `destroy()` function.

PDFPlugin inherits `deviceScaleFactor()` from ScrollableArea, and since UnifiedPDFPlugin
may not end up being a ScrollableArea we can avoid multiple inheritance issues by just
changing code in PluginView to avoid calling it on the plugin.

PluginView creates a UnifiedPDFPlugin based on the runtime flag now.

Fix Unified sources fallout.

* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStoreCollection.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.h:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
(isType):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::PDFPlugin):
(WebKit::PDFPlugin::setView):
(WebKit::PDFPlugin::teardown):
(WebKit::PDFPlugin::destroy): Deleted.
(WebKit::PDFPlugin::isFullFramePlugin const): Deleted.
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
(WebKit::PDFPluginBase::isUnifiedPDFPlugin const):
(WebKit::PDFPluginBase::isLegacyPDFPlugin const):
(WebKit::PDFPluginBase::willDetachRenderer):
(WebKit::PDFPluginBase::isComposited const):
(WebKit::PDFPluginBase::geometryDidChange):
(WebKit::PDFPluginBase::visibilityDidChange):
(WebKit::PDFPluginBase::contentsScaleFactorChanged):
(WebKit::PDFPluginBase::updateControlTints):
(WebKit::PDFPluginBase::wantsWheelEvents const):
(WebKit::PDFPluginBase::isBeingDestroyed const):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::PDFPluginBase):
(WebKit::PDFPluginBase::setView):
(WebKit::PDFPluginBase::destroy):
(WebKit::PDFPluginBase::isFullFramePlugin const):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h: Added.
(isType):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm: Added.
(WebKit::UnifiedPDFPlugin::create):
(WebKit::UnifiedPDFPlugin::UnifiedPDFPlugin):
(WebKit::UnifiedPDFPlugin::teardown):
(WebKit::UnifiedPDFPlugin::scaleFactor const):
(WebKit::UnifiedPDFPlugin::pdfDocumentForPrinting const):
(WebKit::UnifiedPDFPlugin::pdfDocumentSizeForPrinting const):
(WebKit::UnifiedPDFPlugin::streamDidReceiveResponse):
(WebKit::UnifiedPDFPlugin::streamDidReceiveData):
(WebKit::UnifiedPDFPlugin::streamDidFinishLoading):
(WebKit::UnifiedPDFPlugin::streamDidFail):
(WebKit::UnifiedPDFPlugin::liveResourceData const):
(WebKit::UnifiedPDFPlugin::handleMouseEvent):
(WebKit::UnifiedPDFPlugin::handleWheelEvent):
(WebKit::UnifiedPDFPlugin::handleMouseEnterEvent):
(WebKit::UnifiedPDFPlugin::handleMouseLeaveEvent):
(WebKit::UnifiedPDFPlugin::handleContextMenuEvent):
(WebKit::UnifiedPDFPlugin::handleKeyboardEvent):
(WebKit::UnifiedPDFPlugin::handleEditingCommand):
(WebKit::UnifiedPDFPlugin::isEditingCommandEnabled):
(WebKit::UnifiedPDFPlugin::getSelectionString const):
(WebKit::UnifiedPDFPlugin::existingSelectionContainsPoint const):
(WebKit::UnifiedPDFPlugin::rectForSelectionInRootView const):
(WebKit::UnifiedPDFPlugin::countFindMatches):
(WebKit::UnifiedPDFPlugin::findString):
(WebKit::UnifiedPDFPlugin::performDictionaryLookupAtLocation):
(WebKit:: const):
(WebKit::UnifiedPDFPlugin::snapshot):
(WebKit::UnifiedPDFPlugin::accessibilityHitTest const):
(WebKit::UnifiedPDFPlugin::accessibilityObject const):
(WebKit::UnifiedPDFPlugin::accessibilityAssociatedPluginParentForElement const):
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::createPlugin):
(WebKit::PluginView::PluginView):
(WebKit::PluginView::initializePlugin):
(WebKit::PluginView::platformLayer const):
(WebKit::PluginView::scroll):
(WebKit::PluginView::scrollPositionForTesting const):
(WebKit::PluginView::horizontalScrollbar):
(WebKit::PluginView::verticalScrollbar):
(WebKit::PluginView::wantsWheelEvents):
(WebKit::PluginView::paint):
(WebKit::PluginView::invalidateRect):
* Source/WebKit/WebProcess/Plugins/PluginView.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.mm:

Canonical link: https://commits.webkit.org/268476@main

mattwoodrow committed:
Composited mix-blend-mode inside backdrop-filter doesn't use the backdrop-filter as its composited group.
https://bugs.webkit.org/show_bug.cgi?id=149553
<rdar://22854172>

Reviewed by Simon Fraser.

Backdrop-filter creates a stacking context (and isolated group for blend modes), so we should add child layers
onto the primary layer, not as siblings of the backdrop layer.

* LayoutTests/css3/filters/backdrop/backdrop-filter-with-composited-blend-mode-expected.html: Added.
* LayoutTests/css3/filters/backdrop/backdrop-filter-with-composited-blend-mode.html: Added.
* Source/WebCore/platform/graphics/ca/GraphicsLayerCA.cpp:
(WebCore::GraphicsLayerCA::updateSublayerList):

Canonical link: https://commits.webkit.org/268477@main

bls1999 committed:
[Gardening]: REGRESSION(268154@main?): [ macOS ] TestWebKitAPI.WKNavigation.WebProcessLimit is a constant failure.
rdar://116069794
https://bugs.webkit.org/show_bug.cgi?id=262134

Unreviewed test gardening.

Disabling constantly failing API test.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebContentProcessDidTerminate.mm:
(TEST):

Canonical link: https://commits.webkit.org/268478@main

alanbaradlay committed:
[IFC][Ruby] Place inter-character annotation box
https://bugs.webkit.org/show_bug.cgi?id=262129

Reviewed by Antti Koivisto.

Inter-character annotation boxes are placed to the right of the base and vertically centered (don't support 'ruby-align: start').

* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::InlineDisplayContentBuilder::appendAtomicInlineLevelDisplayBox):
(WebCore::Layout::InlineDisplayContentBuilder::handleInlineBoxEnd):
(WebCore::Layout::InlineDisplayContentBuilder::appendIntercharacterRubyAnnotationBox):
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.h:
* Source/WebCore/layout/formattingContexts/inline/ruby/RubyFormattingContext.cpp:
(WebCore::Layout::RubyFormattingContext::annotationPosition):

Canonical link: https://commits.webkit.org/268479@main

hyjorc1 committed:
Add fast path of getting arguments.length for LLInt and Baseline
https://bugs.webkit.org/show_bug.cgi?id=261543
rdar://115462355

Reviewed by Yusuke Suzuki.

We observed that `arguments.length` is frequently used for reading only
without accessing/modifying arguments self or its other properties. If that's
the case we should directly get `argumentCountIncludingThis` from CallFrame
without materializing `arguments` object first. So far, DFG and FTL can
handle this well in varargs forwarding phase. This patch adds the fast
path of getting arguments.length for LLInt and Baseline with simple
heuristic in parser layer.

JavaScriptCode:

    function test() {
        return arguments.length === 3;
    }

Bytecode Before:

    Predecessors: [ ]
    [   0] enter
    [   1] create_direct_arguments dst:loc5
    [   3] mov                dst:loc6, src:loc5
    [   6] get_by_id          dst:loc7, base:loc6, property:0, valueProfile:1
    [  12] stricteq           dst:loc7, lhs:loc7, rhs:Int32: 3(const0)
    [  16] ret                value:loc7
    Successors: [ ]

Bytecode After:

    Predecessors: [ ]
    [   0] enter
    [   1] argument_count     dst:loc5
    [   3] stricteq           dst:loc5, lhs:loc5, rhs:Int32: 3(const0)
    [   7] ret                value:loc5
    Successors: [ ]

Micro Benchmark Result:

                                without                     with

get-arguments-length-fast    2.6024+-0.0750     ^      2.4034+-0.0359        ^ definitely 1.0828x faster

* JSTests/microbenchmarks/get-arguments-length-fast.js: Added.
(test):
* JSTests/stress/get-arguments-length-fast.js: Added.
(shouldBe):
(argumentsWithNestedNormalFunction.nested):
(argumentsWithNestedNormalFunction):
(argumentsNonLengthAccess):
(argumentsLengthAssignment):
(argumentsAssignmentLHS):
(argumentsAssignmentRHS):
(argumentsDestructuringAssignment):
(declareVarArguments):
(declareVarArgumentsWithBlock):
(declareLexicalArguments):
(declareLexicalArgumentsWithBlock):
(declareArrowFunctionArguments):
(declareArrowFunctionArgumentsWithBlock):
(declareFunctionArguments.arguments):
(declareFunctionArguments):
(declareFunctionArgumentsWithBlock.arguments):
(declareFunctionArgumentsWithBlock):
(declareParameterArguments):
(returnArguments):
(helper):
(passArgumentsToFunctionCall):
(ifArguments):
(forInArguments):
(forInVarArguments):
(forInLetArguments):
(forOfArguments):
(forOfVarArguments):
(forOfLetArguments):
(withArguments):
(argumentsWithNestedFunctions.nestedNormalFunction):
(argumentsWithNestedFunctions):
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp:
(JSC::BytecodeGenerator::BytecodeGenerator):
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.h:
(JSC::BytecodeGenerator::useGetArgumentsLengthFast):
* Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp:
(JSC::DotAccessorNode::emitBytecode):
* Source/JavaScriptCore/parser/ASTBuilder.h:
(JSC::ASTBuilder::createDotAccess):
(JSC::ASTBuilder::createAssignment):
(JSC::ASTBuilder::createAssignmentElement):
(JSC::ASTBuilder::checkAssignmentForArguments):
(JSC::ASTBuilder::useArgumentsOnlyForReadingLength):
(JSC::ASTBuilder::usesArguments):
* Source/JavaScriptCore/parser/Nodes.h:
(JSC::ExpressionNode::isArgumentsAccess const):
(JSC::ExpressionNode::isArgumentsLengthAccess const):
(JSC::ExpressionNode::isArguments const):
(JSC::ScopeNode::useGetArgumentsLengthFast const):
* Source/JavaScriptCore/parser/Parser.cpp:
(JSC::Parser<LexerType>::parseInner):
* Source/JavaScriptCore/parser/Parser.h:
(JSC::Scope::hasDeclaredArguments const):
(JSC::Scope::setHasDeclaredArguments):
(JSC::Scope::declareCallee):
(JSC::Scope::declareVariable):
(JSC::Scope::declareFunction):
(JSC::Scope::declareLexicalVariable):
(JSC::Scope::addDeclaredVariable):
(JSC::Scope::addDeclareLexicalVariable):
(JSC::Scope::addDeclaredParameters):
(JSC::Scope::declareParameter):
(JSC::Scope::getSloppyModeHoistedFunctions):
(JSC::Scope::setReadArgumentsLengthOnly):
(JSC::Scope::useGetArgumentsLengthFast const):
(JSC::Parser::popScopeInternal):
* Source/JavaScriptCore/parser/ParserModes.h:
* Source/JavaScriptCore/parser/VariableEnvironment.h:

Canonical link: https://commits.webkit.org/268480@main

rniwa committed:
REGRESSION(266877@main): [ macOS wk2 ] requestidlecallback/requestidlecallback-in-page-cache.html is a flaky text failure/timeout.
https://bugs.webkit.org/show_bug.cgi?id=261341

Reviewed by Wenson Hsieh.

The flakiness was caused by the check sometimes running before idle callbacks.
Delay the check using idle callbacks to enforce the ordering.

* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/requestidlecallback/requestidlecallback-in-page-cache.html:

Canonical link: https://commits.webkit.org/268481@main

bls1999 committed:
New Test (268092@main): [ Release ] WKWebExtensionAPITabs.RemovedEvent is flaky (sometimes wrong tab ID)
rdar://115809590
https://bugs.webkit.org/show_bug.cgi?id=261848

Unreviewed test gardening.

Disabling flaky API test.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITabs.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/268482@main

bls1999 committed:
[Gardening]: REGRESSION(267193@main): [ Debug ] TestWebKitAPI.WebKit2.CrashGPUProcessWhileCapturingAndCalling is a near-constant failure.
rdar://116083315
https://bugs.webkit.org/show_bug.cgi?id=262145

Unreviewed test gardening.

Disabling consistently failing API test.

* Tools/TestWebKitAPI/Tests/WebKit/GetUserMedia.mm:

Canonical link: https://commits.webkit.org/268483@main

hortont424 committed:
Rename CGDisplayList and friends to their real names
https://bugs.webkit.org/show_bug.cgi?id=261938
rdar://115880900

Reviewed by Wenson Hsieh.

Rename and upstream various dynamic-content-scaling-related classes.
Also, remove a very noisy bit of logging, and a setting that is no longer needed.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/ImageBuffer.h:
* Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.h:
* Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.mm:
(-[CALayer _web_clearContents]):
(-[CALayer _web_clearDynamicContentScalingDisplayListIfNeeded]):
(-[CALayer _web_clearCGDisplayListIfNeeded]): Deleted.
* Source/WebCore/platform/graphics/re/DynamicContentScalingTypes.h: Copied from Source/WebCore/platform/graphics/cocoa/WebCoreCALayerExtras.h.
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingDisplayList.cpp: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp.
(WebKit::DynamicContentScalingDisplayList::encode):
(WebKit::DynamicContentScalingDisplayList::decode):
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingDisplayList.h: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h.
(WebKit::DynamicContentScalingDisplayList::DynamicContentScalingDisplayList):
(WebKit::DynamicContentScalingDisplayList::buffer const):
(WebKit::DynamicContentScalingDisplayList::takeSurfaces):
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingImageBufferBackend.h: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.h.
* Source/WebKit/Shared/RemoteLayerTree/DynamicContentScalingImageBufferBackend.mm: Renamed from Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.mm.
(WTF::CFTypeTrait<CAMachPortRef>::typeID):
(WebKit::makeContextOptions):
(WebKit::GraphicsContextDynamicContentScaling::GraphicsContextDynamicContentScaling):
(WebKit::DynamicContentScalingImageBufferBackend::calculateMemoryCost):
(WebKit::DynamicContentScalingImageBufferBackend::create):
(WebKit::DynamicContentScalingImageBufferBackend::DynamicContentScalingImageBufferBackend):
(WebKit::m_renderingMode):
(WebKit::DynamicContentScalingImageBufferBackend::createBackendHandle const):
(WebKit::DynamicContentScalingImageBufferBackend::context):
(WebKit::DynamicContentScalingImageBufferBackend::backendSize const):
(WebKit::DynamicContentScalingImageBufferBackend::bytesPerRow const):
(WebKit::DynamicContentScalingImageBufferBackend::releaseGraphicsContext):
(WebKit::DynamicContentScalingImageBufferBackend::copyNativeImage):
(WebKit::DynamicContentScalingImageBufferBackend::createNativeImageReference):
(WebKit::DynamicContentScalingImageBufferBackend::getPixelBuffer):
(WebKit::DynamicContentScalingImageBufferBackend::putPixelBuffer):
(WebKit::DynamicContentScalingImageBufferBackend::debugDescription const):
(WebKit::DynamicContentScalingAcceleratedImageBufferBackend::create):
(WebKit::DynamicContentScalingAcceleratedImageBufferBackend::DynamicContentScalingAcceleratedImageBufferBackend):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::clearBackingStore):
(WebKit::hasValue):
(WebKit::RemoteLayerBackingStore::encode const):
(WebKit::RemoteLayerBackingStoreProperties::decode):
(WebKit::RemoteLayerBackingStore::swapToValidFrontBuffer):
(WebKit::RemoteLayerBackingStore::supportsPartialRepaint const):
(WebKit::RemoteLayerBackingStore::ensureFrontBuffer):
(WebKit::RemoteLayerBackingStore::paintContents):
(WebKit::RemoteLayerBackingStore::drawInContext):
(WebKit::RemoteLayerBackingStoreProperties::layerContentsBufferFromBackendHandle):
(WebKit::RemoteLayerBackingStoreProperties::applyBackingStoreToLayer):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm:
(WebKit::RemoteLayerTreePropertyApplier::applyPropertiesToLayer):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::replayDynamicContentScalingDisplayListsIntoBackingStore const):
(WebKit::RemoteLayerTreeHost::replayCGDisplayListsIntoBackingStore const): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/cocoa/RemoteLayerTreeLayers.h:
* Source/WebKit/UIProcess/RemoteLayerTree/cocoa/RemoteLayerTreeLayers.mm:
(-[WKCompositingLayer _setWKContents:withDisplayList:replayForTesting:]):
(-[WKCompositingLayer drawInContext:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandle.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/PlatformCALayerRemote.mm:
(WebKit::PlatformCALayerRemote::shouldIncludeDisplayListInBackingStore const):
(WebKit::PlatformCALayerRemote::updateBackingStore):
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.h:
(WebKit::RemoteLayerTreeContext::useDynamicContentScalingDisplayListsForDOMRendering const):
(WebKit::RemoteLayerTreeContext::setUseDynamicContentScalingDisplayListsForDOMRendering):
(WebKit::RemoteLayerTreeContext::useCGDisplayListsForDOMRendering const): Deleted.
(WebKit::RemoteLayerTreeContext::setUseCGDisplayListsForDOMRendering): Deleted.
(WebKit::RemoteLayerTreeContext::useCGDisplayListImageCache const): Deleted.
(WebKit::RemoteLayerTreeContext::setUseCGDisplayListImageCache): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::updatePreferences):

Canonical link: https://commits.webkit.org/268484@main

whsieh committed:
Move off of UIKit SPI: UIWebFormAccessory and UIWebFormAccessoryDelegate
https://bugs.webkit.org/show_bug.cgi?id=262121

Reviewed by Tim Horton.

Stop using this SPI class and delegate protocol; instead, introduce `WKFormAccessoryView` and
`WKFormAccessoryViewDelegate`, which are (nearly) drop-in replacements for `UIWebFormAccessory`.
The only functionality on `UIWebFormAccessory` which we drop in our new WebKit class is support for
showing a "Clear" button (`-setClearVisible:`), which was only necessary when using the "wheels of
time" UI in the keyboard on iPhone; this hasn't been necessary since the form control refresh,
which moved date pickers into a context menu presentation (and more recently, into popover UI).

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:

Remove now-unused SPI declarations.

* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKFormInputSession accessoryViewCustomButtonTitle]):
(-[WKContentView accessoryDone]):
(-[WKContentView accessoryViewDone:]):
(-[WKContentView accessoryTab:]):
(-[WKContentView accessoryView:tabInDirection:]):

Instead of using the existing `-accessoryTab:` delegate that takes a `BOOL`, we can instead use an
`enum class` to represent the direction (next or previous) for slightly better clarity.

(-[WKContentView accessoryViewAutoFill:]):
(-[WKContentView formAccessoryView]):
(-[WKContentView _updateAccessory]):
(-[WKContentView accessoryClear]): Deleted.
(-[WKContentView accessoryAutoFill]): Deleted.

Make `_formAccessoryView` a `WKFormAccessoryView` instead of a `UIWebFormAccessory`, and conform to
`WKFormAccessoryViewDelegate` instead of `UIWebFormAccessoryDelegate`.

* Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.h: Added.
* Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.mm: Added.
(WebKit::doneButtonFont):
(WebKit::singleLineAutoFillButtonFont):
(WebKit::multipleLineAutoFillButtonFont):
(WebKit::upArrow):
(WebKit::downArrow):
(WebKit::createToolbarWithItems):
(-[WKFormAccessoryView _initForUniversalControlBar:]):
(-[WKFormAccessoryView initWithInputAssistantItem:delegate:]):
(-[WKFormAccessoryView traitCollectionDidChange:]):
(-[WKFormAccessoryView _updateFrame]):
(-[WKFormAccessoryView layoutSubviews]):
(-[WKFormAccessoryView _done]):
(-[WKFormAccessoryView _previousTapped]):
(-[WKFormAccessoryView _nextTapped]):
(-[WKFormAccessoryView _autoFill]):
(-[WKFormAccessoryView _autoFillButton]):
(-[WKFormAccessoryView _refreshAutofillPresentation]):
(-[WKFormAccessoryView _applyTreatmentToAutoFillLabel]):
(-[WKFormAccessoryView hideAutoFillButton]):
(-[WKFormAccessoryView showAutoFillButtonWithTitle:]):
(-[WKFormAccessoryView autoFillButtonItem]):
(-[WKFormAccessoryView setNextPreviousItemsVisible:]):
(-[WKFormAccessoryView setNextEnabled:]):
(-[WKFormAccessoryView isNextEnabled]):
(-[WKFormAccessoryView setPreviousEnabled:]):
(-[WKFormAccessoryView isPreviousEnabled]):
(-[WKFormAccessoryView contentRatio]):
* Source/WebKit/UIProcess/mac/DisplayCaptureSessionManager.mm:
(WebKit::DisplayCaptureSessionManager::promptForGetDisplayMedia):

Unrelated unified source build fix.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/ios/UIKitSPI.h:

Canonical link: https://commits.webkit.org/268485@main

bls1999 committed:
[Gardening]: REGRESSION(268408@main): [ macOS ] TestWebKitAPI.WebPushDPushNotificationEventTest.Basic is constantly timing out.
rdar://116087660
https://bugs.webkit.org/show_bug.cgi?id=262150

Unreviewed test gardening.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm:

Canonical link: https://commits.webkit.org/268486@main

ryanhaddad commented on Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Dashboard.js:
```diff
@@ -25,20 +25,21 @@
-        LinuxGTK: { name : "linux-gtk", readableName: "Linux GTK", order: 78 },
-        LinuxJSCOnly: { name : "linux-jsconly", readableName: "Linux JSCOnly", order: 79 },
-        PlayStation: { name : "playstation", readableName: "PlayStation", order: 87 },
+        macOSSonoma: { name: "macos-sonoma", readableName: "macOS Ventura", order: 1 },
```
```suggestion
        macOSSonoma: { name: "macos-sonoma", readableName: "macOS Sonoma", order: 1 },
```

ryanhaddad reviewed (approved):
r+ with comment assuming you've tested the dashboard loads locally.

beidson committed:
Replace CapabilityValueOrRange with CapabilityRange
https://bugs.webkit.org/show_bug.cgi?id=262099
rdar://116038461

Reviewed by Eric Carlson.

- Remove basically unused "value()" feature
- Do nothing by store ranges internall
- Update all users of the class to be nicer
- Generate encode/decode, with validated min/max

* Source/WebCore/Modules/mediastream/MediaTrackCapabilities.cpp:
(WebCore::capabilityDoubleRange):
(WebCore::capabilityIntRange):
* Source/WebCore/platform/mediastream/RealtimeMediaSource.cpp:
(WebCore::RealtimeMediaSource::supportsSizeFrameRateAndZoom):
(WebCore::RealtimeMediaSource::fitnessDistance):
(WebCore::RealtimeMediaSource::applyConstraint):
(WebCore::RealtimeMediaSource::extractVideoFrameSizeConstraints):
(WebCore::RealtimeMediaSource::applyConstraints):
* Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h:
(WebCore::CapabilityValueOrRange::CapabilityValueOrRange):
(WebCore::CapabilityValueOrRange::rangeMin const):
(WebCore::CapabilityValueOrRange::rangeMax const):
(WebCore::CapabilityValueOrRange::value const):
(WebCore::CapabilityValueOrRange::encode const): Deleted.
(WebCore::CapabilityValueOrRange::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/268487@main

beidson committed:
Generate serializers for FontTaggedSettings.h
https://bugs.webkit.org/show_bug.cgi?id=262138
rdar://116079823

Reviewed by Alex Christensen.

* Source/WebCore/platform/graphics/FontTaggedSettings.h:
(WebCore::FontTaggedSetting<T>::encode const): Deleted.
(WebCore::FontTaggedSetting<T>::decode): Deleted.
(WebCore::FontTaggedSettings<T>::encode const): Deleted.
(WebCore::FontTaggedSettings<T>::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/268488@main
